File: InsertionSort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j = j - 1
        arr[j + 1] = key
    return arr


def insertion_sort_reverse(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1
        while j >= 0 and key > arr[j]:
            arr[j + 1] = arr[j]
            j = j - 1
        arr[j + 1] = key
    return arr


arr = [4, 2, 0, 6, 9]

ascending = insertion_sort(arr)
print(ascending)
descending = insertion_sort_reverse(arr)
print(descending)

File: InsertionSort.cs
using System;
class InsertionSort {
    static void insertionSort(int[] arr) {
        for (int i = 1; i < arr.Length; i++) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && key < arr[j]) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
    }

    static void insertionSortReverse(int[] arr) {
        for (int i = 1; i < arr.Length; i++) {
            int key = arr[i];
            int j = i - 1;
            while (j >= 0 && key > arr[j]) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = key;
        }
    }

    static void Main() {
        int[] arr = new int[] {4, 2, 0, 6, 9};

        insertionSort(arr);
        foreach(int j in arr) {
            Console.Write(j + " ");
        }

        Console.WriteLine();

        insertionSortReverse(arr);
        foreach(int j in arr) {
            Console.Write(j + " ");
        }
    }
}